### PR TITLE
Adding missing macros loggers

### LINKED ISSUE #1485: KRATOS_DETAIL and KRATOS_TRACE macros are missing
imagen

As the tittle says the corresponding macros (described in the wiki https://github.com/KratosMultiphysics/Kratos/wiki/How-to-use-Logger) are missing

### PR DESCRIPTION
**Pull request by loumalouomega**
this fixes #1485 (I'm just editing in the comment so that the issue is auto-closed once you merge this).

### CODE REVIEW

File: kratos/input_output/logger.h [hunk 1]
```diff
@@ -251,6 +251,23 @@ namespace Kratos
...
+#define KRATOS_DETAIL_FIRST_N(label, logger_count) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES < logger_count) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::DETAIL
+
+#ifdef KRATOS_DEBUG
+#define KRATOS_TRACE(label) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_IF(label, conditional) if(conditional) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_ONCE(label) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES == __COUNT__) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_FIRST_N(label, logger_count) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES < logger_count) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#else
+#define KRATOS_TRACE(label) if(false) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_IF(label, conditional) if(false) if(conditional) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
```
Comment (roigcarlo) [inline kratos/input_output/logger.h]: merge if's

File: kratos/input_output/logger.h [hunk 2]
```diff
@@ -251,6 +251,23 @@ namespace Kratos
...
+#define KRATOS_TRACE_ONCE(label) static int KRATOS_LOG_OCCURRENCES = -1; if(false) if (++KRATOS_LOG_OCCURRENCES == __COUNT__) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
```
Comment (roigcarlo) [inline kratos/input_output/logger.h]: Same ^

File: kratos/input_output/logger.h [hunk 3]
```diff
@@ -251,6 +251,23 @@ namespace Kratos
...
+#define KRATOS_TRACE_FIRST_N(label, logger_count) static int KRATOS_LOG_OCCURRENCES = -1; if(false) if (++KRATOS_LOG_OCCURRENCES < logger_count) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
```
Comment (roigcarlo) [inline kratos/input_output/logger.h]: Same ^

**Review by roigcarlo — CHANGES_REQUESTED** [review]

**Comment by roigcarlo** [comment]
Actually why don't just define the macros as empty?

**Comment by loumalouomega** [comment]
True, but I looked to the KRATOS_ERROR and I think in replicate structure:

imagen

I can add an empty macros if you prefer

**Comment by roigcarlo** [comment]
@loumalouomega sorry never mind, you need the definition for the << operator. But you can merge the ifs :D

**Comment by loumalouomega** [comment]
I think is easier to define the voids

File: kratos/input_output/logger.h [hunk 4]
```diff
@@ -251,6 +251,23 @@ namespace Kratos
...
+#define KRATOS_DETAIL_ONCE(label) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES == __COUNT__) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::DETAIL
+#define KRATOS_DETAIL_FIRST_N(label, logger_count) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES < logger_count) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::DETAIL
+
+#ifdef KRATOS_DEBUG
+#define KRATOS_TRACE(label) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_IF(label, conditional) if(conditional) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_ONCE(label) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES == __COUNT__) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#define KRATOS_TRACE_FIRST_N(label, logger_count) static int KRATOS_LOG_OCCURRENCES = -1; if (++KRATOS_LOG_OCCURRENCES < logger_count) Logger(label) << KRATOS_CODE_LOCATION << Logger::Severity::TRACE
+#else
+#define KRATOS_TRACE(label) if(false) KRATOS_ERROR
```
Comment (pooyan-dadvand) [inline kratos/input_output/logger.h]: KRATOS_ERROR sends an exception and I won't choose it here to ease the work of optimizer. A KRATOS_WARNING would be better.

**Review by pooyan-dadvand — DISMISSED** [review]

File: kratos/input_output/logger.h (hunk 4 above)
Comment (pooyan-dadvand) [reply kratos/input_output/logger.h]: And look that appveyor saying the same....

Comment (loumalouomega) [reply kratos/input_output/logger.h]: Ok

**Review by roigcarlo — APPROVED** [review]
:+1:

**Pull request merged**